pkg/token/token_writer.go

```diff
@@ -0,0 +1,144 @@
+package token
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/go-logr/logr"
+	v1 "k8s.io/api/core/v1"
+	metaV1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/informers"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/cache"
+
+	"github.com/xenitab/azdo-proxy/pkg/auth"
+)
+
+type TokenWriter struct {
+	logger logr.Logger
+	client *kubernetes.Clientset
+	authz  auth.Authorization
+}
+
+const (
+	usernameValue = "git"
+	usernameKey   = "username"
+	passwordKey   = "password"
+	tokenKey      = "token"
+)
+
+func NewTokenWriter(logger logr.Logger, client *kubernetes.Clientset, authz auth.Authorization) *TokenWriter {
+	return &TokenWriter{
+		logger: logger,
+		client: client,
+		authz:  authz,
+	}
+}
+
+func (t *TokenWriter) Start(stopCh <-chan struct{}) error {
```

Comment:
If we move the logic around stopChan to main, we wouldn't have to pass it here and could handle graceful shutdown of both go routines in main instead of in each.

Example: https://github.com/XenitAB/mqtt-log-stdout/blob/main/cmd/mqtt-log-stdout/main.go#L50-L82